pytorch_lightning/trainer/trainer.py

```diff
@@ -1058,29 +1132,8 @@ def _run_train(self) -> None:
 
         self.reset_train_val_dataloaders(model)
 
-        try:
-            # reset trainer on this loop and all child loops in case user connected a custom loop
-            self.fit_loop.trainer = self
-            self.fit_loop.run()
-        except KeyboardInterrupt:
-            rank_zero_warn("Detected KeyboardInterrupt, attempting graceful shutdown...")
-            # user could press Ctrl+c many times... only shutdown once
-            if not self.interrupted:
-                self.state.status = TrainerStatus.INTERRUPTED
-                self.on_keyboard_interrupt()
-                # same treatment as below
-                self.accelerator.on_train_end()
-        except BaseException:
-            self.state.status = TrainerStatus.INTERRUPTED
-            if distributed_available() and self.world_size > 1:
-                # try syncing remaing processes, kill otherwise
-                self.training_type_plugin.reconciliate_processes(traceback.format_exc())
-            # give accelerators a chance to finish
-            self.accelerator.on_train_end()
-            self._on_exception()
-            # reset bookkeeping
-            self.state.stage = None
-            raise
```

Comment:
@carmocca @awaelchli - before, this error handling was in `_run_train` meaning that it was run as part of the spawned process after the trampoline back to the trainer here: https://github.com/PyTorchLightning/pytorch-lightning/blob/24f0124ddd478e18b76d64a79f1d7b341a0f8646/pytorch_lightning/plugins/training_type/ddp_spawn.py#L201

Do we need similar handling here inside of `new_process` ? https://github.com/PyTorchLightning/pytorch-lightning/blob/24f0124ddd478e18b76d64a79f1d7b341a0f8646/pytorch_lightning/plugins/training_type/ddp_spawn.py#L157-L166

Reply:
Good point. Yes I think the error handling needs to happen inside the spawned processes as well :( 

From the start I didn't understand why we need to move the try-catch construct higher up at all. The main motivation in the issue description seems to be to apply error handling in validate/test, but couldn't this be achieved by wrapping the same try-catch around the individual `loop.run` calls like we had in run_train?

Reply:
@awaelchli @daniellepintz in addition to validate/test, the motivation is such that we can catch & log errors that occur before loop.run (ie. misconfiguration errors, errors during setup). for example, the keyboard interrupt could be applied before loop.run, but this means the exception handling is not kicked in until after the dispatch

this way, one could handle *all* errors that happen after the trainer takes control of the runtime

Reply:
I tried to add error handling around https://github.com/PyTorchLightning/pytorch-lightning/blob/24f0124ddd478e18b76d64a79f1d7b341a0f8646/pytorch_lightning/plugins/training_type/ddp_spawn.py#L201 as per @ananthsub's suggestion, but am still getting this error P438437067 for my newly added test `test_error_handling_all_stages`. The error is being successfully caught (see logging "GOT A DUMMY EXCEPTION"), but somehow the test isn't recognizing that? I wonder if it is an issue with `pytest.raises` although that seems unlikely

Reply:
@daniellepintz the paste isn't available outside FB. you can use https://gist.github.com/ to share the output

Reply:
@daniellepintz - the error handling rethrows the original exception. when run with a plugin that spawns the processes, this leads to a `ProcessRaisedException` 
you would need to update the test to catch this instead

Reply:
thanks so much Ananth! done.